django/db/backends/base/schema.py

```diff
@@ -282,6 +288,27 @@ def _column_default_sql(self, field):
         """
         return '%s'
 
+    def db_default_sql(self, field):
+        """Return the sql and params for the field's database default."""
+        query = Query(model=field.model)
+        compiler = query.get_compiler(connection=self.connection)
+        default_sql, params = compiler.compile(field.db_default)
+        wrapper_sql = self._wrapper_sql_for_default(field.db_default)
+        return self._format_sql_for_default(wrapper_sql, default_sql, params)
+
+    def _wrapper_sql_for_default(self, default):
+        """
+        Return the wrapper sql required for database defaults.
+
+        Database functions require a slightly different syntax to literals.
+        """
+        from django.db.models.expressions import Value
```

Comment:
Would be nice to avoid this import and the one in `django.db.models.fields`. Not sure if it makes sense here but while implementing support for indexes on expressions I [implemented a subclass](https://github.com/django/django/blob/main/django/db/models/indexes.py#L191) of `Expression` which handled stuff like this. I'm thinking something like `DefaultExpression(Expression)` which can handle this parentheses wrapping, special backend handling with `as_mysql` and wrapping of non-expressions in `Value`. 

Might be hard to avoid an import against `django.db.models.expressions` in `django.db.models.fields` but we would at least get rid of this one.

Reply:
I don't think I understand how this would be used well enough to see the value.